src/main/java/ai/lucidtech/las/sdk/Credentials.java

```diff
@@ -32,39 +40,101 @@ public Credentials(String credentialsPath) {
     /**
      *  Used to fetch and store credentials.
      *
-     * @param accessKeyId Access Key Id
-     * @param secretAccessKey Secret Access Key
+     * @param clientId Client id
+     * @param clientSecret Client secret
      * @param apiKey API key
+     * @param authEndpoint Auth endpoint
+     * @param apiEndpoint API endpoint
      */
-    public Credentials(String accessKeyId, String secretAccessKey, String apiKey) {
-        this.accessKeyId = accessKeyId;
-        this.secretAccessKey = secretAccessKey;
+    public Credentials(String clientId, String clientSecret, String apiKey, String authEndpoint, String apiEndpoint) {
+        this.clientId = clientId;
+        this.clientSecret = clientSecret;
         this.apiKey = apiKey;
+        this.authEndpoint = authEndpoint;
+        this.apiEndpoint = apiEndpoint;
+    }
+
+    public String getApiKey() {
+        return apiKey;
+    }
+
+    private Map readCredentialsFromEnviron() {
+        Map<String, String> pairs = new HashMap<String, String>();
+        pairs.put("clientId", "LAS_CLIENT_ID");
+        pairs.put("clientSecret", "LAS_CLIENT_SECRET");
+        pairs.put("apiKey", "LAS_API_KEY");
+        pairs.put("authEndpoint", "LAS_AUTH_ENDPOINT");
+        pairs.put("apiEndpoint", "LAS_API_ENDPOINT");
+
+        Map<String, String> result = new HashMap<String, String>();
+
+        for (Map.Entry<String, String> entry : pairs.entrySet()) {
+            result.put(entry.getKey(), System.getenv(entry.getValue()));
+        }
+
+        return result;
     }
 
     private void readCredentials(String credentialsPath) {
         // TODO Read INI files properly
         try(FileInputStream input = new FileInputStream(credentialsPath)) {
             Properties properties = new Properties();
             properties.load(input);
+            Map<String, String> credentialsFromEnviron = this.readCredentialsFromEnviron();
 
-            this.accessKeyId = properties.getProperty("access_key_id");
-            this.secretAccessKey = properties.getProperty("secret_access_key");
+            for (Object prop : properties.entrySet()) {
+
+            }
```

Comment:
What's happening here?

Reply:
It's for environment variables support, just not finished yet.